app/controllers/uploads_controller.rb

```diff
@@ -39,6 +41,18 @@ def set_page
     @page = [params[:page].to_i, 1].max
   end
 
+  def set_upload
+    @upload = Upload.find_by(id: params[:id])
+    return render nothing: true if @upload.blank?
+  end
+
+  def check_filename
+    filename = [ params[:filename], params[:format] ].compact.join('.')
+    if filename != @upload.file.file.filename
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.